core/jvm/src/main/scala/org/typelevel/idna4s/core/CodePointPlatform.scala

```diff
@@ -0,0 +1,28 @@
+/*
+ * Copyright (c) 2022 Typelevel
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy of
+ * this software and associated documentation files (the "Software"), to deal in
+ * the Software without restriction, including without limitation the rights to
+ * use, copy, modify, merge, publish, distribute, sublicense, and/or sell copies of
+ * the Software, and to permit persons to whom the Software is furnished to do so,
+ * subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in all
+ * copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS
+ * FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR
+ * COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER
+ * IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN
+ * CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+ */
+
+package org.typelevel.idna4s.core
+
+private[core] trait CodePointPlatform {
+
+  private[core] def nameForCodePoint(codePoint: CodePoint): Option[String] =
+    Some(Character.getName(codePoint.value))
```

Comment:
The JavaDoc says:

> Returns the Unicode name of the specified character codePoint, or null if the code point is [unassigned](https://docs.oracle.com/javase/8/docs/api/java/lang/Character.html#UNASSIGNED).

I think only pre-validated inputs are going into this anyway, but I don't think it would hurt to swap `Some(...)` for `Option(...)`.

Reply:
Good catch, I'll update that and add a test for it.